docs/python_binding.rst

```diff
@@ -117,9 +117,30 @@ To get a python dictionary of the settings, call ``session::get_settings``.
 
 .. _`library reference`: reference.html
 
-Retrieving session statistics in Python is more convenient than that in C++.
-The statistics are stored as an array in ``session_stats_alert``, which will be posted after calling ``post_session_stats()`` in the ``session`` object.
-In order to interpret the statistics array, in C++ it is required to call ``session_stats_metrics()`` to get the indices of these metrics, while in Python it can be done using ``session_stats_alert.values["NAME_OF_METRIC"]``, where ``NAME_OF_METRIC`` is the name of a metric.
+Retrieving session statistics in Python is more convenient than that in C++. The
+statistics are stored as an array in ``session_stats_alert``, which will be
+posted after calling ``post_session_stats()`` in the ``session`` object. In
+order to interpret the statistics array, in C++ it is required to call
+``session_stats_metrics()`` to get the indices of these metrics, while in Python
+it can be done using ``session_stats_alert.values["NAME_OF_METRIC"]``, where
+``NAME_OF_METRIC`` is the name of a metric.
+
+set_alert_notify
+================
+
+The ``set_alert_notify()`` function is not compatible with python. Since it
+requires locking the GIL from within the libtorrent thread, to call the callback,
+it can cause a deadlock with the main thread.
+
+Instead, use the python-specific ``set_alert_fd()`` which takes a file descriptor
+that will have 1 byte written to it to notify the client that there are new
+alerts to be popped.
+
+This can be used with ``socket.socketpair()``, for exanple. The file descriptor
```

Comment:
I'd mention that `os.pipe` is also fine and even preferred for UNIX-like systems, due to being simpler and lower-overhead object. It's bad on Windows due to not being able to be configured non-blocking. though.

Reply:
I don't think the non-blocking is a serious issue, and I would think `os.pipe` would probably be a reasonable cross-platform solution